packages/ui/src/ui/pages/query-tracker/module/query_aco/selectors.ts

```diff
@@ -24,6 +26,19 @@ export const getQueryACOOptions = (state: RootState): SelectOption[] => {
 export const isSupportedQtACO = (state: RootState) =>
     getState(state).data.supported_features?.access_control;
 
+export const selectIsMultipleAco = (state: RootState) =>
+    getState(state).data.supported_features?.multiple_aco;
+
+export const selectAccessControlObjects = (state: RootState) =>
+    getState(state).data.access_control_objects;
+
 export const isQueryTrackerInfoLoading = (state: RootState) => getState(state).loading;
 
 export const getQueryTrackerInfo = (state: RootState) => getState(state).data;
+
+export const isSupportedShareQuery = createSelector(
+    [selectIsMultipleAco, selectAccessControlObjects],
+    (isMultipleAco, aco) => {
+        return Boolean(isMultipleAco) && aco.includes(SHARED_QUERY_ACO);
```

Comment:
`aco.includes(SHARED_QUERY_ACO)`-part looks suspiciously here.
Doesn't it mean such query is shared? If so it should not affect visibility of `Share`/`Unshare` button.

Reply:
the fact is that we are moving from a regular aco field to an aco array. Both variants of this field are currently being sent in the request. As a string and as an array. At the moment, the array has a higher priority. So I check for a specific rule in the array